### PR TITLE
chore: add cypress, jest and testing library

### PR DESCRIPTION
**Pull request by glebapps**
## ✍️ Description
- Add cypress, jest and testing library. For the e2e tests, we've created a separated project to isolate it, to avoid dependency issues.
- Update workflows to run the tests and separate them, one for ci and one for e2e test.
- Group components by folder.
- Update the component imports, by creating and `index.ts` file inside the components' folder, which contains all the imports. Then, to use any component, you just need to do it like so `import { Button } from '@/components'`. This way, if we need to do any refactor (change name, location, etc.) you just need to do it once and in one single file.

## ✅ QA

Before requesting a review, please make sure that:

- [x] Preview is working on Netlify, Heroku or similar.
- [x] Manually check that it's working.
- [x] Add documentation and tests if needed.

## 📸 Screenshots
Screenshot 2021-07-26 at 12 32 06
Screenshot 2021-07-26 at 12 32 49

## 🔗 Relevant URLs
- [Task](https://app.clickup.com/t/nnfubn)

### CODE REVIEW

File: components/button/Button.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+describe('Button', () => {
+  it('executes the action when button is click', () => {
+    const onClick = jest.fn()
+    render(<Button label="save" onClick={onClick} />)
+
+    fireEvent.click(screen.getByText('save'))
+
+    expect(onClick).toHaveBeenCalled()
+  })
+})
```
Comment (josepjaume) [inline components/button/Button.spec.tsx:16]: Have you considered this? https://storybook.js.org/docs/react/workflows/unit-testing

As it turns out, you can easily do this to reuse data in stories:

```javascript
//👇 Imports a specific story for the test
import { Primary } from './Button.stories';

it('renders the button in the primary state', () => {
  render(<Primary {...Primary.args} />);
  expect(screen.getByRole('button')).toHaveTextContent('Primary');
});
```

File: next.config.js [hunk 2]
```diff
@@ -14,6 +14,10 @@ const nextConfig = {
   sassOptions: {
     prependData: `@import "styles/variables"; @import "styles/mixins";`,
   },
+  images: {
+    loader: 'imgix',
+    path: 'https://noop/',
+  },
```
Comment (josepjaume) [inline next.config.js:20]: I don't think this is right? It will break production for sure 🤔

**Review by josepjaume — CHANGES_REQUESTED** [review]
Nice job! Requesting a couple of changes though ;)

File: next.config.js (hunk 2 above)
Comment (josepjaume) [reply next.config.js:20]: It fails on the Preview environment as well:

image

**Comment by josepjaume** [comment]
💬  As a very last suggestion (we don't have do to this now!) we could also add Storybook's snapshot testing as well!

https://storybook.js.org/docs/react/workflows/snapshot-testing

File: components/button/Button.spec.tsx (hunk 1 above)
Comment (glebapps) [reply components/button/Button.spec.tsx:16]: Interesting, I didn't know you could that.

But I think we will mainly cover testing with e2e tests, and some specific parts will be covered by unit tests, and unit tests tend to test logic more than the UI. Also, for tests, you want different data for different scenarios. So, not sure what's the benefit here. 🤔

File: next.config.js (hunk 2 above)
Comment (glebapps) [reply next.config.js]: I've just removed and I remembered why we added it, because if we don't, then we can't generate the site, `npm run export` fails:
Screenshot 2021-07-26 at 12 58 21

Comment (josepjaume) [reply next.config.js]: That's right! With Next, if we want to use the `Image` component, we can't just do `npm export`. That's why we're actually using `npm run build` and leveraging Netlify workers.

Next is a bit interesting as it automatically determines if a page needs to use Server Side Rendering or not depending on the hook you use to fetch the data: https://nextjs.org/docs/basic-features/pages#two-forms-of-pre-rendering

So, long story short: We can just use `npm run build` instead of `npm run export` and it will basically do the same as far as we use (`getStaticProps`)[https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation]

Comment (glebapps) [reply next.config.js]: But for the e2e tests we need the generated site.

If you take a look at the e2e workflow, first we generate the site, then we serve it and we run the tests on the generated site:
Screenshot 2021-07-26 at 13 13 50

Comment (josepjaume) [reply next.config.js]: Oh...! This is going to be tricky then. Could we spin un a server instead? Should be doable with `npm run start`.

**Comment by josepjaume** [comment]
Hi @glebapps! I've been talking to @beagleknight about this. Let me see if I can make a proposal in this very PR ;)

**Review by josepjaume — APPROVED** [review]

**Comment by josepjaume** [comment]
There you go @glebapps - fixed! I had to approve my own contribuion though 😆
